### PR TITLE
feat(android): enable colors for status & navigation bar

### PR DESCRIPTION
**Pull request by janpio**
Variation of https://github.com/apache/cordova-plugin-splashscreen/pull/124:

> I have improved it a little with color in navigation bar too: goinnn@7e5bafd

https://github.com/apache/cordova-plugin-splashscreen/pull/124#issuecomment-400614963

### CODE REVIEW

**Comment by goinnn** [comment]
Thanks @janpio. With this approach you can do something like this:

Customizing status bar & footer bar. Nice ;-)

splash

File: src/android/SplashScreen.java [hunk 1]
```diff
@@ -307,12 +311,45 @@ public void run() {
...
                 splashDialog = new Dialog(context, android.R.style.Theme_Translucent_NoTitleBar);
+                splashWindow = splashDialog.getWindow();
                 // check to see if the splash screen should be full screen
                 if ((cordova.getActivity().getWindow().getAttributes().flags & WindowManager.LayoutParams.FLAG_FULLSCREEN)
                         == WindowManager.LayoutParams.FLAG_FULLSCREEN) {
                     splashDialog.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                             WindowManager.LayoutParams.FLAG_FULLSCREEN);
                 }
+
+                // Inspirated in https://github.com/apache/cordova-plugin-splashscreen/pull/124/files
```
Comment (SchnWalter) [inline src/android/SplashScreen.java:322]: No need to have this in code, it's already in the PR.

P.S. Thanks for the work.

File: src/android/SplashScreen.java [hunk 2]
```diff
@@ -307,12 +311,45 @@ public void run() {
...
+                if (statusBarColor != null && !statusBarColor.isEmpty() && Build.VERSION.SDK_INT >= 19) {
+
+                    splashWindow.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+                    splashWindow.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+                    try {
+                        // Using reflection makes sure any 5.0+ device will work without having to compile with SDK level 21
+                        splashWindow.getClass().getDeclaredMethod("setStatusBarColor", int.class).invoke(splashWindow, Color.parseColor(statusBarColor));
+                    } catch (Exception ignore) {
+                        // this should not happen, only in case Android removes this method in a version > 21
+                        LOG.w("SplashScreen StatusBarColor", "Method window.setStatusBarColor not found for SDK level " + Build.VERSION.SDK_INT);
```
Comment (SchnWalter) [inline src/android/SplashScreen.java:334]: I've tested this inside a Android emulators and it works fine on API 27, 28 and 29.

P.S. But it doesn't work on devices with EMUI 10 and API v29:
```
SplashScreen StatusBarColor: »»» USING COLOR #ffd51e2b
SplashScreen StatusBarColor: Method window.setStatusBarColor not found for SDK level 29
SplashScreen StatusBarColor: »»» java.lang.NoSuchMethodException: com.android.internal.policy.HwPhoneWindow.setStatusBarColor

Comment (timbru31) [reply src/android/SplashScreen.java:334]: Given that we’ve dropped support for SDK <=21, we could remove this reflection hack altogether and just call the method normally. The next version of the splashscreen will be a major, too, so no problem to drop Android <=5.0 support here,too.

File: src/android/SplashScreen.java [hunk 3]
```diff
@@ -382,4 +419,4 @@ public void run() {
             }
         });
     }
-}
+}
```
Comment (timbru31) [inline src/android/SplashScreen.java:422]:
```suggestion
}

```

**Comment by Kraloz** [comment]
What is the state of this PR?

**Comment by timbru31** [comment]
There are open review comments that need to be addressed before the PR is mergable.

**Comment by SidiBecker** [comment]
Hello! Any release forecast?

**Comment by 1antares1** [comment]
Any change or workaround for Ionic? : /

**Pull request closed without merging**